packages/components/filters/src/filters.tsx

```diff
@@ -1,22 +1,223 @@
+     */
+    onSortRequest?: Function;
+  };
+  /**
+   * optional key to group filters together.
+   */
+  groupKey?: string;
+  /**
+   * indicates whether filter menu can be removed from filters
+   */
+  isPersistent?: boolean;
+  /**
+   * whether or not the filter is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * unique identifier for the filter
+   */
+  key: string;
+  /**
+   * formatted message to display the filter name
+   */
+  label: ReactNode;
+};
+
+type TFilterGroupConfiguration = {
+  /**
+   * unique identifier for the filter group
+   */
+  key: string;
+  /**
+   * formatted message to display the filter group name
+   */
+  label: ReactNode;
+};
 
 export type TFiltersProps = {
   /**
-   * This is a stub prop
+   * array of applied filters, each containing a unique key and an array of values.
    */
-  label?: string;
+  appliedFilters: TAppliedFilter[];
+  /**
+   * configuration for the available filters.
+   */
+  filters: TFilterConfiguration[];
+  /**
+   * optional configuration for filter groups.
+   */
+  filterGroups?: TFilterGroupConfiguration[];
+  /**
+   * controls the `clear all` (added filters) button from the menu list
+   */
+  onClearAllRequest: () => void;
+  /**
+   * optional callback trigger to add a new filter.
+   */
+  onAddFilterRequest?: Function;
+  /**
+   * function to render a search input, selectable from applicable UI Kit components.
+   */
+  renderSearchComponent: () => ReactNode;
+};
+
+const horizontalDividerStyles = css`
+  width: 100%;
+  height: 1px;
+  border: 1px solid ${designTokens.colorNeutral90};
+  margin-top: ${designTokens.spacing25};
+  margin-bottom: ${designTokens.spacing30};
+`;
+
+const verticalDividerStyles = css`
+  width: 1px;
+  height: ${designTokens.spacing30};
+  background-color: ${designTokens.colorNeutral90};
+  margin: 0 ${designTokens.spacing30} 0 ${designTokens.spacing30}!important;
+`;
+
+const FiltersButton = ({ onAddFilters }: { onAddFilters: () => void }) => {
+  const intl = useIntl();
+  return (
+    <FlatButton
+      data-testid="filters-button"
+      label={intl.formatMessage(messages.filtersButtonLabel)}
+      icon={<FilterIcon />}
+      onClick={onAddFilters}
+    />
+  );
+};
+
+const AddFilterButton = () => {
+  const intl = useIntl();
+  return (
+    <FlatButton
+      data-testid="add-filter-button"
+      label={intl.formatMessage(messages.addFiltersButtonLabel)}
+      icon={<PlusBoldIcon />}
+    />
+  );
+};
+
+//Removes filters from being visible & clears all data
+const ClearAllFiltersButton = ({
+  onClearAllRequest,
+}: {
+  onClearAllRequest: () => void;
+}) => {
+  const intl = useIntl();
+  return (
+    <>
+      <div css={verticalDividerStyles}> </div>
+      <FlatButton
+        icon={<CloseBoldIcon />}
+        label={intl.formatMessage(messages.clearAllFiltersButtonLabel)}
+        onClick={onClearAllRequest}
+        tone="secondary"
+      />
+    </>
+  );
```

Comment:
is it necessary to declare these buttons as separate components?  since they arent reused, it seems like they could be declared directly in the parent component.  Personally, I'd find that more readable.

If others disagree, my care amount here is fairly low.